.github/workflows/check-query-files-and-compilation.yml

```diff
@@ -1,37 +1,83 @@
 name: Parser compilation and query file check
```

Comment:
Why are the artifacts uploaded in this job? Are they used for something?

Reply:
I don't know; I believe there might be people (or workflows) who don't want to build their own parsers?

Reply:
Doing these on each commit seems like a waste 🤔 

![image](https://user-images.githubusercontent.com/8050659/150569442-ebf9dfaa-1b31-4d62-a4d1-3595fac3ce69.png)

Reply:
Agreed.

Reply:
Anyway, it's not related to this PR. I was just curious.

Reply:
It was mostly for debugging because sometimes the CI would build successfully artifacts that wouldn't work. The Windows wiki has a hint that users can download the parsers from CI. But it would probably be better to have binary releases for that.

Reply:
Let me know if I should remove that step.

Reply:
>Doing these on each commit seems like a waste thinking

Well, we're not paying the servers :smile: was wondering how much data we caused by that. They seem to be persistent for a pretty long time. I was expecting that at some point we'd either hit API limits or that stuff get's quickly ejected.

Reply:
Probably, they should be replaced by releases. I was experimenting a bit with that https://github.com/theHamsta/nvim-treesitter/releases/tag/nightly-parsers releases would also have a predictable path for download (e.g. tagged releases of our plugin could auto-download the binaries for the platform if a user opts in). But even better would be to do that outside of our org.

Reply:
> Well, we're not paying the servers

I meant "waste" as in "doing unnecessary work repeatedly as it is producing and uploading the same files for each commit" 😂 I'm sure github removes old files once it hits the limit.

Reply:
or ghcr.io ?

Reply:
(+1 on releases, but not sure why outside of the org would be better?)

Reply:
Because tree-sitter 1.0 wants to offer an easy/standardized way for parsers to publish without having generated files in their repo. But sure for us it would make sense to have all PRs

Reply:
Oh, I see. Yes, if upstream took care of this, that'd be cake :)

Reply:
@clason See here how releases would work https://github.com/theHamsta/nvim-treesitter/commit/a0e29464e75bc1cf146523379d2d274c6f09d4b6 . You can only do this on tags because there you can hit API limits fairly quickly :smile:

Reply:
Yeah, we would probably want to do a daily job or something for that.

Anyway, all that is for a different PR -- is there anything I need to change for this one? I'm just waiting for your approval ;)